From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sat, 17 Dec 2022 17:11:38 +0100
Subject: build: Add a pkgconfig file with plugin information
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

This allows other tools and external pugins to figure out
the file system locations.

Signed-off-by: Guido Günther <guido.gunther@puri.sm>
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1168>
---
 plugins/meson.build | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/plugins/meson.build b/plugins/meson.build
index 031a0b6..8b91c77 100644
--- a/plugins/meson.build
+++ b/plugins/meson.build
@@ -1,3 +1,14 @@
 subdir('calendar')
 subdir('upcoming-events')
 subdir('ticket-box')
+
+pkgconfig.generate(
+  variables : ['lockscreen_plugins_dir=@0@'.format(plugins_dir),
+              ],
+  subdirs : 'phosh',
+  version : meson.project_version(),
+  install_dir : libdir / 'pkgconfig',
+  filebase : 'phosh-plugins',
+  name : 'phosh-plugins',
+  description : 'Phosh plugin configuration data',
+)
